File: leetcode/1935. 可以输入的最大单词数.cpp
class Solution {
public:
    int canBeTypedWords(string text, string brokenLetters) {
        set<int> s;
        for (int i = 0; i < text.size(); ++i)
        {
            if (text[i] == ' ')
            {
                s.insert(i);
            }
        }
        int res = 0;
        s.insert(text.size());
        for (auto i : s)
        {
            for (int j = i - 1; j >= 0 && text[j] != ' '; --j)
            {
                if (brokenLetters.find(text[j]) != string::npos)
                {
                    res++;
                    break;
                }
            }
        }
        return s.size() - res;
    }
};

File: leetcode/1935. 可以输入的最大单词数.rs
impl Solution {
    pub fn can_be_typed_words(text: String, broken_letters: String) -> i32 {
        use std::collections::HashSet;
        let broken: HashSet<char> = broken_letters.chars().collect();

        let mut count = 0;
        for word in text.split_whitespace() {
            if !word.chars().any(|c| broken.contains(&c)) {
                count += 1;
            }
        }
        count
    }
}